upcloud_api/cloud_manager/server_mixin.py

```diff
@@ -111,6 +111,7 @@ def create_server(self, server: Server) -> Server:
           * Debian 10: 01000000-0000-4000-8000-000020050100
```

Comment:
Could also update these templates. At least, Debian and Ubuntu require metadata service to be available so might need some additional docs changes as well 🤔

```
Debian GNU/Linux 12 (Bookworm): 01000000-0000-4000-8000-000020070100
Ubuntu Server 24.04 LTS (Noble Numbat): 01000000-0000-4000-8000-000030240200
Rocky Linux 9: 01000000-0000-4000-8000-000150020100
Windows Server 2022 Standard: 01000000-0000-4000-8000-000010080300
```

Or we could just replace the listing with: Use `upctl storage list --public --template` to list available public templates.

Reply:
My guess is that `metadata` defaults to `False`, so when using those new templates that require cloud-init, user would have to configure `metadata=True` when creating the server 🤔

Reply:
ok, updated the code in server creation to force metadata to be true if the user chose an OS which requires it.